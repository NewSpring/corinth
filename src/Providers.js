import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import { LiveProvider } from '@apollosproject/ui-connected';

import NavigationService from './NavigationService';
import ClientProvider from './client';
import ExternalLinkProvider from './linking/Provider';
import track from './analytics';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      navigate={NavigationService.navigate}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() => NavigationService.navigate('Onboarding')}
      >
        <ExternalLinkProvider navigate={NavigationService.navigate}>
          <MediaPlayerProvider>
            <AnalyticsProvider trackFunctions={[track]}>
              <LiveProvider>
                <Providers
                  themeInput={customTheme}
                  iconInput={customIcons}
                  {...props}
                />
              </LiveProvider>
            </AnalyticsProvider>
          </MediaPlayerProvider>
        </ExternalLinkProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
