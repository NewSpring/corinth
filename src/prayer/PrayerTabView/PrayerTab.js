import React, { memo } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import {
  styled,
  Button,
  BodyText,
  H4,
  Placeholder,
  FlexedView,
} from '@apollosproject/ui-kit';

const VerticalPaddedView = styled(({ theme }) => ({
  paddingVertical: theme.sizing.baseUnit,
}))(View);

const ContentView = styled(({ theme }) => ({
  paddingHorizontal: theme.sizing.baseUnit,
  marginBottom: theme.sizing.baseUnit * 2,
}))(FlexedView);

const StyledButton = styled({
  width: '100%',
})(Button);

const PrayerTab = memo(
  ({ prayers, type, title, description, loading, ...props }) => (
    <ContentView>
      <Placeholder.Paragraph
        lineNumber={2}
        onReady={!loading}
        lastLineWidth="100%"
        firstLineWidth="60%"
      >
        {prayers.length > 0 ? (
          <View>
            <VerticalPaddedView>
              <BodyText placeholder={'Loading Prayers'}>{description}</BodyText>
            </VerticalPaddedView>
            <StyledButton
              title="Start praying"
              onPress={() =>
                props.navigation.navigate('PrayerList', {
                  prayers,
                  title,
                })
              }
            />
          </View>
        ) : (
          <VerticalPaddedView>
            {type === 'saved' ? (
              <H4>You do not have any saved prayers</H4>
            ) : (
              <>
                <H4>There are no prayers yet for your {type}</H4>
                <BodyText>Be the first to add one!</BodyText>
              </>
            )}
          </VerticalPaddedView>
        )}
      </Placeholder.Paragraph>
    </ContentView>
  )
);

PrayerTab.propTypes = {
  description: PropTypes.string,
  title: PropTypes.string,
  type: PropTypes.string,
  prayers: PropTypes.arrayOf(
    PropTypes.shape({}) // TODO fill this out
  ),
  loading: PropTypes.bool,
};

PrayerTab.displayName = 'PrayerTab';

export default PrayerTab;
