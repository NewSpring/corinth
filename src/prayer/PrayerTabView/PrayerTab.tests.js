import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'newspringchurchapp/src/Providers';
import PrayerTab from './PrayerTab';

it('renders prayer PrayerTab', () => {
  const tree = renderer.create(
    <Providers>
      <PrayerTab prayers={[]} navigation={jest.fn()} />
    </Providers>
  );
  expect(tree).toMatchSnapshot();
});

it('renders prayer tab with content', () => {
  const tree = renderer.create(
    <Providers>
      <PrayerTab
        navigation={jest.fn()}
        prayers={[]}
        description={'Pray for your saved prayers'}
        title={'My Saved Prayers'}
        type={'saved'}
      />
    </Providers>
  );
  expect(tree).toMatchSnapshot();
});
