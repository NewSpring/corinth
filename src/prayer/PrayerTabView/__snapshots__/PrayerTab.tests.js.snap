// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders prayer PrayerTab 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginBottom": 32,
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#303030",
          "fontFamily": "Colfax-Bold",
          "fontSize": 16,
          "lineHeight": 23.04,
        }
      }
    >
      There are no prayers yet for your 
    </Text>
    <Text
      bold={false}
      italic={false}
      style={
        Object {
          "color": "#303030",
          "fontFamily": "Colfax-Regular",
          "fontSize": 16,
          "lineHeight": 26,
        }
      }
    >
      Be the first to add one!
    </Text>
  </View>
</View>
`;

exports[`renders prayer tab with content 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginBottom": 32,
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#303030",
          "fontFamily": "Colfax-Bold",
          "fontSize": 16,
          "lineHeight": 23.04,
        }
      }
    >
      You do not have any saved prayers
    </Text>
  </View>
</View>
`;
